exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/QueryManager.java

```diff
@@ -338,7 +345,8 @@ private QueryProfile getQueryProfile(UserException ex) {
         .setStart(startTime)
         .setEnd(endTime)
         .setTotalFragments(fragmentDataSet.size())
-        .setFinishedFragments(finishedFragments.get());
+        .setFinishedFragments(finishedFragments.get())
+        .addAllOptions(queryOptions);
```

Comment:
I have updated code to serialize and deserialize options as you suggested. Now they are added as json string (optionsJson, as in examples you mentioned). Though profile json representation contains escapes for such string, which is probably is not as nice as was before. But it's the only way to display json string inside of json.
![image](https://cloud.githubusercontent.com/assets/15086720/17456969/887cea94-5bf1-11e6-9846-25bbcba68154.png)
If you want, I can revert the changes back to using prototype buffer option object.